Failure to build julia-1.8.0-beta1 on macOS

Trying to build the 1.8.0 beta1 on macOS as part of Homebrew testing, we're seeing a build failure. The build command is:

`make VERBOSE=1 USE_BINARYBUILDER=0 prefix=/opt/homebrew/Cellar/julia/1.8.0-beta1 sysconfdir=/opt/homebrew/etc USE_SYSTEM_CSL=1 USE_SYSTEM_LIBUNWIND=1 USE_SYSTEM_PCRE=1 USE_SYSTEM_OPENLIBM=1 USE_SYSTEM_BLAS=1 USE_SYSTEM_LAPACK=1 USE_SYSTEM_GMP=1 USE_SYSTEM_MPFR=1 USE_SYSTEM_LIBSUITESPARSE=1 USE_SYSTEM_UTF8PROC=1 USE_SYSTEM_MBEDTLS=1 USE_SYSTEM_LIBSSH2=1 USE_SYSTEM_NGHTTP2=1 USE_SYSTEM_CURL=1 USE_SYSTEM_LIBGIT2=1 USE_SYSTEM_PATCHELF=1 USE_SYSTEM_ZLIB=1 USE_SYSTEM_P7ZIP=1 LIBBLAS=-lopenblas LIBBLASNAME=libopenblas LIBLAPACK=-lopenblas LIBLAPACKNAME=libopenblas USE_BLAS64=0 PYTHON=python3 MACOSX_VERSION_MIN=12 JULIA_CPU_TARGET='generic;cortex-a57;thunderx2t99;armv8.2-a,crypto,fullfp16,lse,rdm' TAGGED_RELEASE_BANNER='Built by Homebrew (v1.8.0-beta1)' install`

and the failure is:

```
/usr/bin/tar -xmUzf /private/tmp/julia-20220312-40562-1hpnzsy/julia-1.8.0-beta1/usr-staging/zlib-cacf7f1d4e3d44d871b605da3b647f07d718623f.tgz -C /private/tmp/julia-20220312-40562-1hpnzsy/julia-1.8.0-beta1/usr
install_name_tool -id @rpath/libz.dylib /private/tmp/julia-20220312-40562-1hpnzsy/julia-1.8.0-beta1/usr/lib/libz.dylib
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: can't open file: /private/tmp/julia-20220312-40562-1hpnzsy/julia-1.8.0-beta1/usr/lib/libz.dylib (No such file or directory)
make[2]: *** [/private/tmp/julia-20220312-40562-1hpnzsy/julia-1.8.0-beta1/usr/manifest/zlib] Error 1
make[1]: *** [julia-deps] Error 2
make: *** [install] Error 2
```

Full build output available there: https://gist.github.com/fxcoudert/4c0dc2b0a383dffb8e88f6286a895fd2

It seems to be building `zlib` (and then not finding it), while we requested `USE_SYSTEM_ZLIB=1`

--- a/deps/llvm.mk
+++ b/deps/llvm.mk
@@ -233,10 +233,14 @@ $$(LLVM_BUILDDIR_withtype)/build-compiled: $$(SRCCACHE)/$$(LLVM_SRC_DIR)/$1.patc
 LLVM_PATCH_PREV := $$(SRCCACHE)/$$(LLVM_SRC_DIR)/$1.patch-applied
 endef
 
+ifeq ($(USE_SYSTEM_ZLIB), 0)
+$(LLVM_BUILDDIR_withtype)/build-configured: | $(build_prefix)/manifest/zlib
+endif
+
 # NOTE: LLVM 12 and 13 have their patches applied to JuliaLang/llvm-project
 
 # declare that all patches must be applied before running ./configure
-$(LLVM_BUILDDIR_withtype)/build-configured: | $(LLVM_PATCH_PREV) $(build_prefix)/manifest/zlib
+$(LLVM_BUILDDIR_withtype)/build-configured: | $(LLVM_PATCH_PREV)
 
 $(LLVM_BUILDDIR_withtype)/build-configured: $(SRCCACHE)/$(LLVM_SRC_DIR)/source-extracted | $(llvm_python_workaround)
 	mkdir -p $(dir $@)

